bagel/cli.py

```diff
@@ -329,9 +339,169 @@ def bids(
+    integrate them in a single .jsonld with subjects' harmonized phenotypic data (from the bagel pheno command) and optionally,
+    BIDS metadata (from the bagel bids command).
+    NOTE: Must be run AFTER the pheno command.
+
+    This command will create a valid, subject-level instance of the Neurobagel
+    graph data model for the combined metadata in the .jsonld format.
+    You can upload this .jsonld file to the Neurobagel graph.
+    """
+    futil.check_overwrite(output, overwrite)
+
+    space = 51
+    print(
+        "Running initial checks of inputs...\n"
+        f"   {'Existing subject graph data to augment (.jsonld):' : <{space}}{jsonld_path}\n"
+        f"   {'Processing status file (.tsv):' : <{space}}{tabular}"
+    )
+
+    # Load and do basic TSV validation of the processing status file
+    status_df = futil.load_tabular(tabular, input_type="processing status")
+
+    # Check for missing participant IDs
+    if row_indices := putil.get_rows_with_empty_strings(
+        status_df, [PROC_STATUS_COLS["participant"]]
+    ):
+        raise LookupError(
+            f"Your processing status file contains missing values in the column '{PROC_STATUS_COLS['participant']}'. "
+            "Please ensure that every row has a non-empty participant id. "
+            f"We found missing values in the following rows (first row is zero): {row_indices}."
+        )
+
+    pipelines = status_df[PROC_STATUS_COLS["pipeline_name"]].unique()
+    dutil.check_pipelines_are_recognized(pipelines)
+
+    # Per pipeline, check that version(s) are from the allowed set
+    # TODO: Do we need to check all versions across all pipelines first, and report all unrecognized versions together?
+    for pipeline in pipelines:
+        versions = status_df[
+            status_df[PROC_STATUS_COLS["pipeline_name"]] == pipeline
+        ][PROC_STATUS_COLS["pipeline_version"]].unique()
+
+        dutil.check_pipeline_versions_are_recognized(pipeline, versions)
+
+    jsonld = futil.load_json(jsonld_path)
+    jsonld_dataset = extract_and_validate_jsonld_dataset(jsonld, jsonld_path)
+
+    # Extract subjects from the JSONLD
+    jsonld_subject_dict = extract_subs_from_jsonld_dataset(jsonld_dataset)
+
+    # Check that all subjects in the processing status file are found in the JSONLD
+    unique_derivatives_subs = get_subs_missing_from_pheno_data(
+        subjects=status_df[PROC_STATUS_COLS["participant"]].unique(),
+        pheno_subjects=jsonld_subject_dict.keys(),
+    )
+    if len(unique_derivatives_subs) > 0:
+        raise LookupError(
+            "The specified processing status file contains subject IDs not found in "
+            "the provided json-ld file:\n"
+            f"{unique_derivatives_subs}\n"
+            "Subject IDs are case sensitive. "
+            "Please check that the processing status file corresponds to the dataset in the provided .jsonld."
+        )
+
```

Comment:
The main reason for this refactoring was because the error messages are command-specific. That said, I can adopt the approach we used to refactor `load_tabular()` and try to create a common error message template, and just replace file titles in the string via params. It gives us a bit less flexibility but should work for now !